### PR TITLE
build: refactor cmake files

### PR DESCRIPTION
**Pull request by dundargoc**
- build: rely on default cmake installation if possible

The default cmake installation process for dependencies that use cmake
on all platforms is in general more robust and less verbose, so we rely
on that if possible.

- build(cmake): add modelines to enable syntax highlighting

- build: give example on complex regexes

This is just to allow the reader to get a quick understanding without
necessarily needing to know all the regex intricasies.

### CODE REVIEW

File: cmake.deps/CMakeLists.txt [hunk 1]
```diff
@@ -146,6 +140,20 @@ endif()
 include(ExternalProject)
 set_directory_properties(PROPERTIES EP_PREFIX "${DEPS_BUILD_DIR}")
 
+set(GLOBAL_CMAKE_ARGS
```
Comment (justinmk) [inline cmake.deps/CMakeLists.txt:143]: if these are only used by deps, let's use the `DEPS_` prefix, which already has precedent.
```suggestion
set(DEPS_CMAKE_ARGS
```

Comment (dundargoc) [reply cmake.deps/CMakeLists.txt:143]: Good call.

**Comment by dundargoc** [comment]
Apparently `CMAKE_ARGS` is ignored if `CONFIGURE_COMMAND` is specified, which essentially nullifies any benefits from this PR.

File: cmake.deps/cmake/LibiconvCMakeLists.txt [hunk 2]
```diff
@@ -95,3 +95,5 @@ install(TARGETS libcharset libiconv iconv
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+# vim: set ft=cmake:
```
Comment (justinmk) [inline cmake.deps/cmake/LibiconvCMakeLists.txt:99]: good idea :)

**Review by justinmk — APPROVED** [review]
nice

**Pull request merged**